### PR TITLE
UI series

### PR DESCRIPTION
**Pull request by CS6**
* 頁面 1 的完整功能
* 頁面 2 的 A, B 部分 (包含顯示 series 封面，以及留空的九宮格圖片顯示)
關於頁面 3 的 C 部分  (按照 asset 是否擁有的狀況顯示 asset 或者底圖)，這個部分等 QA 測完第一部分後，會請  @seanwu1105  接手
換句話說，QA 測是你的 release 會期待

- [ ] * 可以從 photo view 切換到 series view 再切回來，都不會有問題，可以正確看到 買過的 series

- [ ] * 點選 series 封面會進入 collection view

- [ ] * 在 collection view 底下，再點選上方的 series view icon 會回到 series view 
 **此部分之前已改為開新頁面顯示**

- [ ] * 在 collection view 底下，可以看到正確的 series 封面與留空的九宮格照片，但裡面目前並無法正確顯示照片

- [ ] * 在 collection view 滑動的時候，封面照片會跟著滑動

* 其他以上未提到的部分，基於 beta13.4

New series 封面樣式
image
image

### CODE REVIEW

**Review by seanwu1105 — CHANGES_REQUESTED** [review]
Make sure to pass `build` job in CI before requesting review.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+})
+export class DiaBackendSeriesRepository {
+  private readonly postCapturesUpdated$ = new Subject<{ reason?: string }>();
+
+  constructor(
+    private readonly httpClient: HttpClient,
+    private readonly authService: DiaBackendAuthService
+  ) {}
+
+  readSeries$() {
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:21]: Rename to `fetchSeries$` for consistency.

Also, as it does NOT have any argument, this method can be a readonly member. For example:

```suggestion
  readonly fetchSeries$ = defer(() => this.authService.getAuthHeaders()).pipe(
    concatMap(headers =>
    ...
    )
 )
```

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return defer(() => this.authService.getAuthHeaders()).pipe(
+      concatMap(headers =>
+        this.httpClient.get<DiaBackendSeries>(`${BASE_URL}/api/v2/series/`, {
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:24]: Incorrect typing. The response `GET` API should be `PaginatedResponse<DiaBackendSeries>`.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        if (err instanceof HttpErrorResponse && err.status === 404) {
+          return throwError(new NotFoundErrorResponse(err));
+        }
+        return throwError(err);
+      })
+    );
+  }
+}
+
+export interface DiaBackendSeries {
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:39]: Most of the fields are unused. Remove the unused fields. Always add the fields when needed.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  readonly id: string;
+  readonly collections: string;
+  readonly in_store: boolean | null | undefined;
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:42]: I believe the API will not return `undefined`. Thus, use `boolean | null` only.
```suggestion
  readonly in_store: boolean | null;
```

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  readonly owner: string;
+  readonly owner_name: string;
+  readonly type: string;
+  readonly cover_image: string | null | undefined;
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:46]: The same as `in_store` property.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]
Apart from the comments in codes, 

- Remove commented codes.
- Use `readonly` if possible.
- Use `undefined` instead of `null` if possible.

File: src/app/features/home/post-capture-tab/post-capture-tab.component.ts [hunk 6]
```diff
@@ -32,12 +32,11 @@ export class PostCaptureTabComponent {
...
-  // collected=True
-  readonly getSeries$ = this.networkService.connected$.pipe(
+  readonly series$ = this.networkService.connected$.pipe(
     switchMap(isConnected =>
       iif(
         () => isConnected,
-        this.diaBackendSeriesRepository.readSeries$().pipe(
+        this.diaBackendSeriesRepository.fetchSeries$().pipe(
           pluck('results')
           // map(series => series.filter(a => a.in_store == true))
```
Comment (seanwu1105) [inline src/app/features/home/post-capture-tab/post-capture-tab.component.ts:41]: Make sure to remove all commented codes as mentioned in the previous change request.

File: src/app/features/home/post-capture-tab/series/series.page.ts [hunk 7]
```diff
@@ -1,7 +1,41 @@
...
   selector: 'app-series',
   templateUrl: './series.page.html',
   styleUrls: ['./series.page.scss'],
 })
-export class SeriesPage {}
+export class SeriesPage implements OnInit {
+  id: string | null | undefined;
+  cover: string | null | undefined;
+  collectionGeneral = [
+    { img: null },
```
[the remark this answers is not in the document]
Comment (seanwu1105) [reply src/app/features/home/post-capture-tab/series/series.page.ts:13]: Replace `null` with `undefined` if possible.

File: src/app/features/home/post-capture-tab/series/series.page.ts [hunk 8]
```diff
@@ -1,14 +1,15 @@
...
   selector: 'app-series',
   templateUrl: './series.page.html',
   styleUrls: ['./series.page.scss'],
 })
-export class SeriesPage implements OnInit {
-  id: string | null | undefined;
-  cover: string | null | undefined;
-  collectionGeneral = [
+export class SeriesPage {
+  readonly id = this.route.snapshot.paramMap.get('id');
```
Comment (seanwu1105) [inline src/app/features/home/post-capture-tab/series/series.page.ts:9]: We do not need `snapshot` to get `paramMap`, use `this.route.paramMap` directly. See the official docs for the details.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
+// tslint:disable: no-magic-numbers
```
[the remark this answers is not in the document]
Comment (seanwu1105) [reply src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:1]: Why mark this "unresolved" converation resolved? Still, you did not remove the deprecated comments for TSLint.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 10]
```diff
@@ -1,32 +1,36 @@
...
+import { PaginatedResponse } from '../pagination';
 import { BASE_URL } from '../secret';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DiaBackendSeriesRepository {
-  private readonly postCapturesUpdated$ = new Subject<{ reason?: string }>();
-
+  private readonly postCapturesCache$ = new ReplaySubject<
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts:15]: Unused property.

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts [hunk 11]
```diff
@@ -1,32 +1,36 @@
...
-        })
+        this.httpClient.get<PaginatedResponse<DiaBackendSeries>>(
+          `${BASE_URL}/api/v2/series/`,
+          {
+            headers,
+          }
+        )
       ),
       catchError((err: unknown) => {
-        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
```
Comment (seanwu1105) [inline src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts]: Why removing comments for ESLint? This is valid, IMHO.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]

File: src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts (hunk 11 above)
Comment (CS6) [reply src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts]: 我是參考這裏，因為目前沒找到比較好的做法😿
https://github.com/numbersprotocol/capture-lite/blob/4833f3a152f522bad70a9beabf6f959ac98353d1/src/app/shared/services/dia-backend/asset/dia-backend-asset-repository.service.ts#L197

Comment (seanwu1105) [reply src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts]: @CS6 是呀，這個是最好的作法，但是你為什麼要把他移除XDD 上面那句英文是說，幹麻移除他，這個是正確的作法呀XD

Comment (CS6) [reply src/app/shared/services/dia-backend/series/dia-backend-series-repository.service.ts]: 我發現我們在不同分支ＱＱ

**Comment by seanwu1105** [comment]
@CS6 Remember to pass all checks before request for review. Thanks.

File: src/app/features/home/post-capture-tab/series/series.page.html [hunk 12]
```diff
@@ -5,26 +5,27 @@
   <div class="toolbar-spacer"></div>
 </mat-toolbar>
 <div class="page-content">
-  <app-series-card></app-series-card>
+  <div class="series-image">
```
Comment (seanwu1105) [inline src/app/features/home/post-capture-tab/series/series.page.html:8]: Use `ion-card` to use built-in card style. Remember to remove unused CSS rules.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]

File: src/app/features/home/post-capture-tab/post-capture-tab.component.html [hunk 13]
```diff
@@ -23,13 +23,18 @@
...
+        <mat-icon class="from-store">shopping_bag</mat-icon>
       </mat-grid-tile>
     </mat-grid-list>
   </div>
 
   <div class="series-wrapper" *ngSwitchCase="'Series'">
-    <app-series-card routerLink="series"></app-series-card>
-    <app-series-card routerLink="series"></app-series-card>
-    <app-series-card routerLink="series"></app-series-card>
+    <div
```
Comment (seanwu1105) [inline src/app/features/home/post-capture-tab/post-capture-tab.component.html:32]: Suggestion Only: Use `ion-card` to use built-in card style. Remember to remove unused CSS rules.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]

**Pull request merged**